src/Symfony/Component/Form/Extension/Core/Type/DayOfWeekType.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+use Symfony\Component\OptionsResolver\Options;
+
+class DayOfWeekType extends AbstractType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    {
+        //  IntlCalendar::DOW_* constants
+        $daysOfWeek = array(
+            \IntlCalendar::DOW_SUNDAY    => 'sunday',
+            \IntlCalendar::DOW_MONDAY    => 'monday',
+            \IntlCalendar::DOW_TUESDAY   => 'tuesday',
+            \IntlCalendar::DOW_WEDNESDAY => 'wednesday',
+            \IntlCalendar::DOW_THURSDAY  => 'thursday',
+            \IntlCalendar::DOW_FRIDAY    => 'friday',
+            \IntlCalendar::DOW_SATURDAY  => 'saturday',
+        );
+
+        /**
+         * Goes through all daysOfWeek once, starting at $firstDayOfWeek
+         *
+         * @param int $firstDayOfWeek
+         * @return array
+         */
+        $getOrderedDaysOfWeek = function ($firstDayOfWeek, $valueFormat) use (&$daysOfWeek) {
+            $days = array();
+
+            // goes through all items once, starting at $firstDayOfWeek
+            for ($dayOfWeek = $firstDayOfWeek - 1; $dayOfWeek < ($firstDayOfWeek + 6); $dayOfWeek++) {
+                $slug = $daysOfWeek[($dayOfWeek % 7) + 1];
+                $value = $valueFormat === 'slug' ?  $slug : $dayOfWeek;
+                $days[$value] = $slug;
+            }
+
+            return $days;
+        };
+
+        /**
+         * Replace values with the formatted day of the week.
+         * Uses strtotime('monday')
+         *
+         * @param IntlDateFormatter $formatter
+         * @param array $days
+         * @return array
+         */
+        $getDayNames = function (\IntlDateFormatter $formatter, array $days) {
+            $names = null;
+            $timestamps = array_map('strtotime', array_values($days));
+            $names = array_map(array($formatter, 'format'), $timestamps);
+
+            return array_combine(array_keys($days), $names);
+        };
+
+        $resolver->setDefault('timezone', null);
```

Comment:
seDefault() is not part of the OptionsResolverInterface. You should collect the values in an array and run a single $resolver->setDefaults() in the end.